js/index.js

```diff
@@ -1,31 +1,49 @@
 // ITERATION 1
 
 function updateSubtotal(product) {
-  console.log('Calculating subtotal, yey!');
+  const price = product.querySelector('.price span');
 
-  //... your code goes here
+  const quantityParent = product.querySelector('.quantity');
+  const quantityChildValue = quantityParent.firstElementChild.value;
+
+  const subtotal = product.querySelector('.subtotal span');
+
+  let priceSubtotal = price.innerText * quantityChildValue;
+  return (subtotal.innerText = priceSubtotal);
 }
 
 function calculateAll() {
-  // code in the following two lines is added just for testing purposes.
-  // it runs when only iteration 1 is completed. at later point, it can be removed.
-  const singleProduct = document.querySelector('.product');
-  updateSubtotal(singleProduct);
-  // end of test
+  // const singleProduct = document.querySelector('.product');
+  // updateSubtotal(singleProduct);
 
-  // ITERATION 2
-  //... your code goes here
+  //ITERATION 2
+  const allProducts = document.getElementsByClassName('product');
+  const allProductsArray = [...allProducts];
+  let allProductsSum = 0;
```

Comment:
are you all copying from each other? :eyes: 

literally nobody used reduce (the right tool for this) and instead this weird .forEach approach. 
```js
const allProductsSum = [...allProducts].reduce((total, product) => total + updateSubtotal(product)), 0)
```